tools/goctl/api/gocligen/utils.go

```diff
@@ -0,0 +1,190 @@
+	"fmt"
+	goformat "go/format"
+	"io"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/zeromicro/go-zero/tools/goctl/api/spec"
+	"github.com/zeromicro/go-zero/tools/goctl/api/util"
+	"github.com/zeromicro/go-zero/tools/goctl/util/ctx"
+	"github.com/zeromicro/go-zero/tools/goctl/util/pathx"
+)
+
+type fileGenConfig struct {
+	dir             string
+	subdir          string
+	filename        string
+	templateName    string
+	category        string
+	templateFile    string
+	builtinTemplate string
+	data            interface{}
+}
+
+func genFile(c fileGenConfig) error {
+	fp, created, err := util.MaybeCreateFile(c.dir, c.subdir, c.filename)
+	if err != nil {
+		return err
+	}
+	if !created {
+		return nil
+	}
+	defer fp.Close()
+
+	var text string
+	if len(c.category) == 0 || len(c.templateFile) == 0 {
+		text = c.builtinTemplate
+	} else {
+		text, err = pathx.LoadTemplate(c.category, c.templateFile, c.builtinTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	t := template.Must(template.New(c.templateName).Parse(text))
+	buffer := new(bytes.Buffer)
+	err = t.Execute(buffer, c.data)
+	if err != nil {
+		return err
+	}
+
+	code := formatCode(buffer.String())
+	_, err = fp.WriteString(code)
+	return err
+}
+
+func getParentPackage(dir string) (string, error) {
+	abs, err := filepath.Abs(dir)
+	if err != nil {
+		return "", err
+	}
+
+	projectCtx, err := ctx.Prepare(abs)
+	if err != nil {
+		return "", err
+	}
+
+	wd := projectCtx.WorkDir
+	d := projectCtx.Dir
+	same, err := pathx.SameFile(wd, d)
+	if err != nil {
+		return "", err
+	}
+
+	trim := strings.TrimPrefix(projectCtx.WorkDir, projectCtx.Dir)
+	if same {
+		trim = strings.TrimPrefix(strings.ToLower(projectCtx.WorkDir), strings.ToLower(projectCtx.Dir))
+	}
+
+	return filepath.ToSlash(filepath.Join(projectCtx.Path, trim)), nil
+}
+
+func formatCode(code string) string {
+	ret, err := goformat.Source([]byte(code))
+	if err != nil {
+		return code
+	}
+
+	return string(ret)
+}
+
+func writeProperty(writer io.Writer, name, tag, comment string, tp spec.Type, indent int) error {
+	util.WriteIndent(writer, indent)
+	var err error
+	if len(comment) > 0 {
+		comment = strings.TrimPrefix(comment, "//")
+		comment = "//" + comment
+		_, err = fmt.Fprintf(writer, "%s %s %s %s\n", strings.Title(name), tp.Name(), tag, comment)
+	} else {
+		_, err = fmt.Fprintf(writer, "%s %s %s\n", strings.Title(name), tp.Name(), tag)
+	}
+
+	return err
+}
+
+func responseGoTypeName(r spec.Route, pkg ...string) string {
+	if r.ResponseType == nil {
+		return ""
+	}
+
+	resp := golangExpr(r.ResponseType, pkg...)
+	switch r.ResponseType.(type) {
+	case spec.DefineStruct:
+		if !strings.HasPrefix(resp, "*") {
+			return "*" + resp
+		}
+	}
+
+	return resp
+}
+
+func getGoTypeName(typ string) string {
+	if !strings.HasPrefix(typ, "*") {
+		return typ
+	}
+	return strings.Replace(typ, "*", "", 1)
+}
+
+func requestGoTypeName(r spec.Route, pkg ...string) string {
+	if r.RequestType == nil {
+		return ""
+	}
+
```

Comment:
There are blank lines here, the method above has no blank lines before return? It would be better to unify